scene/resources/surface_tool.cpp

```diff
@@ -542,7 +542,10 @@ Array SurfaceTool::commit_to_arrays() {
 				for (uint32_t idx = 0; idx < vertex_array.size(); idx++) {
 					const Vertex &v = vertex_array[idx];
 
-					ERR_CONTINUE(v.bones.size() != count);
+					if (v.bones.size() > count) {
```

Comment:
If `v.bones.size() < count`, the indexing in the next `for` loop will segfault, no?

Reply:
What do you mean? 

count is either 4 or 8.

v.bones.size() is a number from 0 to +infinity.

uint32_t idx = 0; idx < vertex_array.size(); idx++

first iteration is:

1. idx = 0
2. Let's assume the bone size is 7
3. 	```for (int j = 0; j < count; j++) {
		w[idx * count + j] = v.bones[j];
	}```
4. idx = idx + 1` from `uint32_t idx = 0; idx < vertex_array.size(); idx++`

The second iteration is:

5. idx is 1
6. Let's assume the bone size is 6
7. ```for (int j = 0; j < count; j++) {
		w[idx * count + j] = v.bones[j];
	}```
8. idx = idex + 1 ` from `uint32_t idx = 0; idx < vertex_array.size(); idx++`

Reply:
Added an `array.fill(0)` line.

Reply:
I think this is correct. Please comment if there are concerns.